network/templates/base/observation_view.html

```diff
@@ -100,7 +105,10 @@
                data-groundstation="{{ data.ground_station }}">
             <div class="panel-heading">
               <h3 class="panel-title">
-                Data payload #{{ data.id }} from {{ data.ground_station }}
+                Data payload #{{ data.id }} from
+                <a href="{% url 'base:station_view' id=data.ground_station.id %}">
+                  {{ data.ground_station }}
+                </a>
```

Comment:
We should probably add some css here to colorfy the link. Currently it's the same color with the text.